OpenSim/Examples/Plugins/VolumetricContactExample/VolumetricContactForce.h

```diff
@@ -0,0 +1,134 @@
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied    *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+//=============================================================================
+// INCLUDES
+//=============================================================================
+// Headers define the various property types that OpenSim objects can read 
+#include <string>
+#include "osimPluginDLL.h"
+#include <OpenSim/Simulation/Model/Force.h>
+#include <OpenSim/Simulation/Model/TwoFrameLinker.h>
+
+
+//=============================================================================
+//=============================================================================
+/*
+ * A class template for writing a custom Force plugin. 
+ * Applies ...
+ * @author         
+ * @version    
+ */
+namespace OpenSim {
+    // TODO: OSIMPLUGIN_API will change to OSEMSIMULATION_API
+class OSIMPLUGIN_API VolumetricContactForce 
+    : public TwoFrameLinker<Force, PhysicalFrame>
+{
+    OpenSim_DECLARE_CONCRETE_OBJECT(VolumetricContactForce, TwoFrameLinker);
+public:
+//=============================================================================
+// PROPERTIES
+//=============================================================================
+    // TODO: look for an osimsimulation library registered type, and attach it to that library instead of a separate one?
+
+    OpenSim_DECLARE_PROPERTY(k_V, double,
+        "Volumetric stiffness (N/m^3)");
+    OpenSim_DECLARE_PROPERTY(a_V, double,
+        "Volumetric damping");
+    OpenSim_DECLARE_PROPERTY(mu_s, double,
+        "Static friction");
+    OpenSim_DECLARE_PROPERTY(mu_d, double,
+        "Dynamic friction");
+    OpenSim_DECLARE_PROPERTY(v_t, double,
+        "Friction transition velocity (m/s)");
+    OpenSim_DECLARE_PROPERTY(w_t, double,
+        "Friction angular transition velocity (rad/s)");
+    OpenSim_DECLARE_PROPERTY(ellipsoid_dims, SimTK::Vec3,
+        "Ellipsoid dimensions (x, y, z axis) (m)");
+
+//=============================================================================
+// METHODS
+//=============================================================================
+public:
+    // Default Constructor
```

Comment:
It would be good to include what the default settings are.

Reply:
For default constructors, what are parameters generally set to? Should I set them to something invalid, so ensure that the user enters values, or perhaps pick arbitrary values?